src/components/sidebar/sidebar.pw.tsx

```diff
@@ -0,0 +1,435 @@
+    mount,
+    page,
+  }) => {
+    await mount(<SidebarComponentFocusable />);
+
+    const toastElement = getComponent(page, "toast");
+
+    await expect(toastElement).not.toBeVisible();
+
+    const openToastElement = getDataElementByValue(page, "open-toast");
+    await openToastElement.click();
+
+    await expect(toastElement).toBeVisible();
+
+    await page.keyboard.press("Tab");
+    const closeIconButtonElement = closeIconButton(page).nth(1);
+
+    await expect(closeIconButtonElement).toBeFocused();
+  });
+
+  test("should call onCancel callback when a click event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <SidebarComponent
+        onCancel={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const closeIconButtonElement = closeIconButton(page);
+    await closeIconButtonElement.click();
+
+    await expect(callbackCount).toEqual(1);
+  });
+
+  test("should ensure the component is rendered on top of any other modals when the topModalOverride prop is true", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<TopModalOverride />);
+
+    const sidebarPreviewElement = sidebarPreview(page);
+
+    await expect(sidebarPreviewElement).toBeVisible();
+
+    const bodyElement = page.locator("body");
+    await bodyElement.press("Tab");
+    const sidebarPreviewCloseIconButton = sidebarPreview(page).locator(
+      CLOSE_ICON_BUTTON
+    );
+
+    await expect(sidebarPreviewCloseIconButton).toBeFocused();
+
+    const focusedElement = page.locator("*:focus");
+    await focusedElement.press("Enter");
+
+    await expect(sidebarPreviewElement).not.toBeVisible();
+
+    await bodyElement.press("Tab");
+    const DialogCloseIconButton = getComponent(page, "dialog").locator(
+      CLOSE_ICON_BUTTON
+    );
+
+    await expect(DialogCloseIconButton).toBeFocused();
+  });
+});
+
+test.describe("Accessibility tests for Sidebar component", () => {
+  test("should check accessibility for SidebarComponent example", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<SidebarComponent />);
+
+    await checkAccessibility(page);
+  });
+
+  (["left", "right"] as SidebarProps["position"][]).forEach((boolVal) => {
+    test(`should check accessibility when sidebar position is ${boolVal}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<SidebarComponent position={boolVal} />);
+
+      await checkAccessibility(page);
+    });
+  });
+
+  test("should check accessibility when sidebar has header", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<SidebarComponentWithHeading />);
+
+    await checkAccessibility(page);
+  });
+
+  ([
+    SIDEBAR_SIZES[0],
+    SIDEBAR_SIZES[1],
+    SIDEBAR_SIZES[2],
+    SIDEBAR_SIZES[3],
+    SIDEBAR_SIZES[4],
+    SIDEBAR_SIZES[5],
+    SIDEBAR_SIZES[6],
+  ] as SidebarProps["size"][]).forEach((size) => {
```

Comment:
**suggestion** (non-blocking): similar to my earlier comment where we used the `SIDEBAR_SIZES` array, but even simpler here - this can be simplified a lot. This array is just a very longhand way of writing `SIDEBAR_SIZES`!